ambari-logsearch/ambari-logsearch-server/src/main/java/org/apache/ambari/logsearch/web/filters/LogsearchTrustedProxyFilter.java

```diff
@@ -0,0 +1,188 @@
+    Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+    boolean skip = true;
+    if (authPropsConfig.isTrustedProxy() && !isAuthenticated(authentication) ) {
+      String doAsUserName = req.getParameter("doAs");
+      String remoteAddr = req.getRemoteAddr();
+      if (StringUtils.isNotEmpty(doAsUserName) && isTrustedProxySever(remoteAddr)
+        && isTrustedHost(getXForwardHeader((HttpServletRequest) req))) {
+        List<GrantedAuthority> grantedAuths = getDefaultGrantedAuthorities();
+        if (!(isTrustedProxyUser(doAsUserName) || isTrustedProxyUserGroup(grantedAuths))) {
+          skip = false;
+        }
+      }
+    }
+    if (skip) {
+      chain.doFilter(req, res);
+      return;
+    }
+    super.doFilter(req, res, chain);
+  }
+
+  private boolean isTrustedProxySever(String requestHosts) {
+    if (authPropsConfig.getProxyIp() == null || requestHosts == null) {
+      return false;
+    }
+    final List<String> proxyServers = authPropsConfig.getProxyIp();
+    return (proxyServers.size() == 1 && proxyServers.contains("*")) || authPropsConfig.getProxyIp().contains(requestHosts);
+  }
+
+  private boolean isTrustedHost(String requestHosts) {
+    if (requestHosts == null) {
+      return false;
+    }
+    List<String> trustedProxyHosts = authPropsConfig.getProxyUserHosts();
+    return (trustedProxyHosts.size() == 1 && trustedProxyHosts.contains("*")) || trustedProxyHosts.contains(requestHosts);
+  }
+
+  private boolean isTrustedProxyUser(String doAsUser) {
+    if (doAsUser == null) {
+      return false;
+    }
+    List<String> trustedProxyUsers = authPropsConfig.getProxyUsers();
+    return (trustedProxyUsers.size() == 1 && trustedProxyUsers.contains("*")) || trustedProxyUsers.contains(doAsUser);
+
+  }
+
+  private boolean isTrustedProxyUserGroup(List<GrantedAuthority> proxyUserGroup) {
+    if (proxyUserGroup == null) {
+      return false;
+    }
+    List<String> trustedProxyGroups = authPropsConfig.getProxyUserGroups();
+    if (trustedProxyGroups.size() == 1 && trustedProxyGroups.contains("*")) {
+      return true;
+    } else {
+      for (GrantedAuthority group : proxyUserGroup) {
+        if (trustedProxyGroups.contains(group.getAuthority())) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  private boolean isAuthenticated(Authentication authentication) {
+    return authentication != null && !(authentication instanceof AnonymousAuthenticationToken) && authentication.isAuthenticated();
+  }
+
+  private String getXForwardHeader(HttpServletRequest httpRequest) {
+    Enumeration<String> names = httpRequest.getHeaderNames();
+    while (names.hasMoreElements()) {
+      String name = names.nextElement();
+      Enumeration<String> values = httpRequest.getHeaders(name);
+      String value = "";
+      if (values != null) {
+        while (values.hasMoreElements()) {
+          value = values.nextElement();
```

Comment:
If only one header exists getHeader could be used.
If there can be more headers with the same name should any of the values checked or just the last one?